examples/misc/logos2.py

```diff
@@ -1,87 +1,157 @@
-            ha='right', va='center', alpha=1.0, transform=ax.transAxes)
-
-
-def add_polar_bar():
-    ax = fig.add_axes([0.025, 0.075, 0.2, 0.85], projection='polar')
-
-    ax.patch.set_alpha(axalpha)
-    ax.set_axisbelow(True)
-    N = 7
-    arc = 2. * np.pi
-    theta = np.arange(0.0, arc, arc/N)
-    radii = 10 * np.array([0.2, 0.6, 0.8, 0.7, 0.4, 0.5, 0.8])
-    width = np.pi / 4 * np.array([0.4, 0.4, 0.6, 0.8, 0.2, 0.5, 0.3])
-    bars = ax.bar(theta, radii, width=width, bottom=0.0)
-    for r, bar in zip(radii, bars):
-        bar.set_facecolor(cm.jet(r/10.))
-        bar.set_alpha(0.6)
-
-    ax.tick_params(labelbottom=False, labeltop=False,
-                   labelleft=False, labelright=False)
-
-    ax.grid(lw=0.8, alpha=0.9, ls='-', color='0.5')
-
-    ax.set_yticks(np.arange(1, 9, 2))
-    ax.set_rmax(9)
-
 
-if __name__ == '__main__':
-    main_axes = add_math_background()
-    add_polar_bar()
-    add_matplotlib_text(main_axes)
-    plt.show()
+    path = TextPath((0, 0), "matplotlib", size=height_px * 0.8,
+                    prop=get_font_properties())
+
+    angle = 4.25  # degrees
+    trans = mtrans.Affine2D().skew_deg(angle, 0)
+
+    patch = PathPatch(path, transform=trans + ax.transData, color=MPL_BLUE,
+                      lw=0)
+    ax.add_patch(patch)
+    ax.autoscale()
+
+
+def make_logo(height_px, lw_bars, lw_grid, lw_border, rgrid, with_text=False):
+    """
+    Create a full figure with the Matplotlib logo.
+
+    Parameters
+    ----------
+    height_px : int
+        Height of the figure in pixel.
+    lw_bars : float
+        The linewidth of the bar border.
+    lw_grid : float
+        The linewidth of the grid.
+    lw_border : float
+        The linewidth of icon border.
+    rgrid : sequence of float
+        The radial grid positions.
+    with_text : bool
+        Whether to draw only the icon or to include 'matplotlib' as text.
+    """
+    dpi = 100
+    height = height_px / dpi
+    figsize = (5 * height, height) if with_text else (height, height)
+    fig = plt.figure(figsize=figsize, dpi=dpi)
+    fig.patch.set_alpha(0)
+
+    if with_text:
+        create_text_axes(fig, height_px)
+    ax_pos = (0.535, 0.12, .17, 0.75) if with_text else (0.03, 0.03, .94, .94)
+    ax = create_icon_axes(fig, ax_pos, lw_bars, lw_grid, lw_border, rgrid)
+
+    return fig, ax
+
+##############################################################################
+# A large logo:
+
+make_logo(height_px=110, lw_bars=0.7, lw_grid=0.5, lw_border=1,
+          rgrid=[1, 3, 5, 7])
+
+##############################################################################
+# A small 32px logo:
+
+make_logo(height_px=32, lw_bars=0.3, lw_grid=0.3, lw_border=0.3, rgrid=[5])
+
+##############################################################################
+# A large logo including text, as used on the matplotlib website.
+
+make_logo(height_px=110, lw_bars=0.7, lw_grid=0.5, lw_border=1,
+          rgrid=[1, 3, 5, 7], with_text=True)
```

Comment:
please add a plt.show() at the end for interactive users.

Reply:
Done.